test/Test/Database/LSMTree/Class/Monoidal.hs

```diff
@@ -362,6 +365,30 @@ evalRange (FromToIncluding lo hi) x = lo <= x && x <= hi
 queryResultKey :: QueryResult k v -> k
 queryResultKey (FoundInQuery k _) = k
 
+queryResultFromLookup :: k -> LookupResult v -> Maybe (QueryResult k v)
+queryResultFromLookup k = \case
+   NotFound -> Nothing
+   Found v -> Just (FoundInQuery k v)
+
+-- | A range lookup behaves like many point lookups.
+prop_lookupRangeLikeLookups ::
+     IsTableHandle h
+  => Proxy h -> [(Key, Update Value)]
+  -> Range Key
+  -> Property
+prop_lookupRangeLikeLookups h ups r = ioProperty $ do
+    withTableNew h ups $ \_ hdl -> do
+      res1 <- rangeLookup hdl r
+
+      let testKeys = V.fromList $ nubSort $ filter (evalRange r) $ map fst ups
+      res2 <- fmap (V.catMaybes . V.zipWith queryResultFromLookup testKeys) $
+        lookups hdl testKeys
+
+      return $ res1 === res2
```

Comment:
This is good but only tests the upper bound: that everything we _do_ get back from `rangeLookup` we can also get from `lookups`. It doesn't check the other lower bound that we get back everything we should get, i.e. all values in the range in the model. We want both bounds.